### PR TITLE
Ishans papp 25839 feature enhancement

### PR DESCRIPTION
**Pull request by ishans-crest**
Attaching integration pipeline link for sanity pipeline as well because this app has asset with very less paste limit and it exhausts on a few runs as well: https://us-west-2.console.aws.amazon.com/codesuite/codebuild/875003031410/projects/app-integration-tests/build/app-integration-tests%3Abee25e82-d6c1-4855-add7-8eb047750a07?region=us-west-2

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: pastebin_connector.py [hunk 1]
```diff
@@ -1,194 +1,538 @@
...
+        self._api_key = ""
+        self._pastebin_username = ""
+        self._pastebin_password = ""
+
+    def _convert_to_client_tz(self, t_str):
+        # TODO: Next version will have a configurable TZ
+        t = parse(t_str, tzinfos={"CDT": -18000, "CST": -21600})
+        client_tz = pytz.UTC
+
+        # try:
```
Comment (splunk-jessica) [inline pastebin_connector.py:64]: Remove comments

File: pastebin_connector.py [hunk 2]
```diff
@@ -1,194 +1,538 @@
...
+        #    client_tz = pytz.timezone("EST5EDT")
+        # except:
+        #    client_tz = pytz.UTC
```
Comment (splunk-jessica) [inline pastebin_connector.py]: Can the commented out code be removed?

File: pastebin_connector.py [hunk 3]
```diff
@@ -1,194 +1,538 @@
...
+
+        try:
+            with open(fname, "w") as outfile:
+                outfile.write(smart_str(file_data))
+        except IOError:
+            success = False
+
+        if success:
+            # Vault.add_attachment(fname, container_id, file_name=None, metadata=dict())
+            # Save attachment file to the vault\
```
Comment (splunk-jessica) [inline pastebin_connector.py]:
```suggestion
            # Save attachment file to the vault
```

File: pastebin_connector.py [hunk 4]
```diff
@@ -1,194 +1,538 @@
...
+        fname = os.path.join(temp_dir, file_name)
+
+        try:
+            with open(fname, "w") as outfile:
+                outfile.write(smart_str(file_data))
+        except IOError:
+            success = False
+
+        if success:
+            # Vault.add_attachment(fname, container_id, file_name=None, metadata=dict())
```
Comment (splunk-jessica) [inline pastebin_connector.py]: Same for this line, can we just remove the commented out code?

File: pastebin_connector.py (hunk 2 above)
Comment (achandya-crest) [reply pastebin_connector.py]: Done.

File: pastebin_connector.py (hunk 3 above)
Comment (achandya-crest) [reply pastebin_connector.py]: Done.

File: pastebin_connector.py (hunk 4 above)
Comment (achandya-crest) [reply pastebin_connector.py]: Done.

**Comment by splunk-jessica** [comment]
looks like a couple of the tests are failing

**Comment by achandya-crest** [comment]
> looks like a couple of the tests are failing

@splunk-jessica The asset has limit in creating pastes that is why the sanity and integration pipelines are failing at the moment due to many pipeline runs. I have already added description in this MR showing 1 passed run of the pipeline.

**Review by splunk-jessica — APPROVED** [review]

**Review by dvora-crest — APPROVED** [review]

**Pull request merged**